test/FlexVotingClient.t.sol

```diff
@@ -354,51 +565,68 @@ contract Vote is FlexVotingClientTest {
     assertEq(_abstainVotesExpressed, _abstainVotesExpressedInit);
   }
 
-  function testFuzz_UsersCannotExpressVotesPriorToDepositing(
-    address _user,
-    uint208 _voteWeight,
-    uint8 _supportType
-  ) public {
-    _voteWeight = _commonFuzzerAssumptions(_user, _voteWeight, _supportType);
+  function testFuzz_RevertOn_UnknownVoteType(address _user, uint208 _voteWeight, uint8 _supportType)
+    public
+  {
+    // Force vote type to be unrecognized.
+    vm.assume(_supportType > uint8(GCF.VoteType.Abstain));
 
-    // Create the proposal *before* the user deposits anything.
-    uint256 _proposalId = _createAndSubmitProposal();
+    vm.assume(_user != address(flexClient));
+    // This max is a limitation of the fractional governance protocol storage.
+    _voteWeight = uint208(bound(_voteWeight, 1, type(uint128).max));
 
     // Deposit some funds.
     _mintGovAndDepositIntoFlexClient(_user, _voteWeight);
 
-    // Now try to express a voting preference on the proposal.
-    assertEq(flexClient.deposits(_user), _voteWeight);
-    vm.expectRevert(bytes("no weight"));
+    // Create the proposal.
+    uint256 _proposalId = _createAndSubmitProposal();
+
+    // Now try to express a voting preference with a bogus support type.
+    vm.expectRevert(bytes("invalid support value, must be included in VoteType enum"));
     vm.prank(_user);
     flexClient.expressVote(_proposalId, _supportType);
   }
+}
 
-  function testFuzz_UsersMustExpressWithKnownVoteType(
-    address _user,
-    uint208 _voteWeight,
-    uint8 _supportType
-  ) public {
-    // Force vote type to be unrecognized.
-    vm.assume(_supportType > uint8(GCF.VoteType.Abstain));
-
-    vm.assume(_user != address(flexClient));
-    // This max is a limitation of the fractional governance protocol storage.
-    _voteWeight = uint208(bound(_voteWeight, 1, type(uint128).max));
+contract CastVote is FlexVotingClientTest {
+  function testFuzz_SubmitsVotesToGovernor(address _user, uint208 _voteWeight, uint8 _supportType)
+    public
+  {
+    _voteWeight = _commonFuzzerAssumptions(_user, _voteWeight, _supportType);
```

Comment:
We should refactor this test helper for a couple reasons, but I will pull that into its own issue since it's not part of what this PR is changing.

Reply:
Decided to just do it in here, see [11fe8d6](https://github.com/ScopeLift/flexible-voting/pull/71/commits/11fe8d67a2e7997ee6cf3b8f378cda67604a3d2d)

Reply:
Nice I like the approach you took